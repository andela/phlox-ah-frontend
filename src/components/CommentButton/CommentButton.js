import React from 'react';
import PropTypes from 'prop-types';
import './CommentButton.scss';

const CommentButton = ({ addComment }) => (
  <div className="comment-button">
    <button onClick={addComment}>Comment</button>
  </div>
);

CommentButton.propTypes = {
  addComment: PropTypes.func,
};

export default CommentButton;
