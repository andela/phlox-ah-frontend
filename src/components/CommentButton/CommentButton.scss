.comment-button {
  display: flex;
  justify-content: flex-end;
  button {
    border: 1px solid #0e718c;
    color: #0e718c;
    font-size: 15px;
    font-weight: 600;
    background-color: #fff;
    padding: 8px 8px;
    outline: none;
    width: 150px;
    cursor: pointer;
    margin-top: 20px;
  }
}