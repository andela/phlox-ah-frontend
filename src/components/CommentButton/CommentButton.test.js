import React from 'react';
import { mount } from 'enzyme';
import CommentButton from './CommentButton';

let component;

beforeEach(() => {
  component = mount(<CommentButton addComment={() => { }}/>);
});

afterEach(() => {
  component.unmount();
});

it('should render without throwing an error', () => {
  expect(component).toMatchSnapshot();
});

it('should have a div', () => {
  expect(component.find('div').length).toEqual(1);
});

it('should have a div with a className of comment-button', () => {
  expect(component.find('div').hasClass('comment-button'));
});

it('should have a button', () => {
  expect(component.find('button').length).toEqual(1);
});

it('should have a button with text Comment', () => {
  expect(component.find('button').text()).toEqual('Comment');
});

it('should have a props of addComment that is a function', () => {
  expect(typeof component.props().addComment).toBe('function');
});
