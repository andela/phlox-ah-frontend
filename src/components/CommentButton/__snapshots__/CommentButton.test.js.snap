// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<CommentButton
  addComment={[Function]}
>
  <div
    className="comment-button"
  >
    <button
      onClick={[Function]}
    >
      Comment
    </button>
  </div>
</CommentButton>
`;
